dart/lib/src/client_reports/discard_reason.dart

```diff
@@ -1,8 +1,5 @@
-import 'package:meta/meta.dart';
-
 /// A reason that defines why events were lost, see
 /// https://develop.sentry.dev/sdk/client-reports/#envelope-item-payload.
-@internal
```

Comment:
`h`: As pointed out by @marandaneto, I think we don't want to make `DiscardReason` or `DiscardedEvent` public.

Reply:
We use this in the Flutter SDK, but we could go with supressing the resulting warning there, like we already do for native frames. WDYT @marandaneto ?

Reply:
Sounds good.

Reply:
We do have the `recordLostEvent` method in [SentryTransport.h](https://github.com/getsentry/sentry-cocoa/blob/a23b2ece0c552468936eee6c139c14843a40ad33/Sources/Sentry/include/SentryTransport.h#L42) in the Cocoa SDK.

The `transport` class is not publicly exposed on Cocoa, but it is in the [Dart SDK](https://github.com/getsentry/sentry-dart/blob/e492fefca2817fbe56e55045b0a0394c50b3eb2c/dart/lib/sentry.dart#L21), I assume because we implement a file based transport in the `Flutter SDK`.

Assuming we implement the same logic as in iOS, we are at least exposing a method and the `DiscardReason`. If we mark them as internal, consumers of the SDK will always have a warning when implementing the transport interface. This would be a problem if it is used by others and not just by ourselves in the Flutter SDK.

Do you have some thought on that @marandaneto ?

Reply:
> We do have the recordLostEvent method in [SentryTransport.h](https://github.com/getsentry/sentry-cocoa/blob/a23b2ece0c552468936eee6c139c14843a40ad33/Sources/Sentry/include/SentryTransport.h#L42) in the Cocoa SDK.

Yes, but on Cocoa, the transport is not yet public.

We could also create an extra class for tracking the discarded events. We could name it, for example, `DiscardedEvents` and both the client and the transport have a reference to it. The discarded events don't have to be directly in the transport. When using a custom transport, the `DiscardedEvents` can't just do nothing, or we could also add a `NoOpDiscardedEvents` or something similar.

Reply:
@philipphofmann Moved the recorder instance out of the transport classes, as the envelopes can be enriched right before passing them to the transport layer. All new classes are now marked as internal again.